fix(lobby): return to welcome screen when rejected

nbeckindy opened the pull request:
If a participant is rejected from the lobby, they end up in a room seemingly alone. This change would send them back to the Welcome screen after a timeout.

jitsi-jenkins commented:
Hi, thanks for your contribution!
If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

damencho commented:
Jenkins test this please

saghul commented:
Not sure if this is enough. On a first thought, I don't think we'll be sending any events to the iframe API for example.

On another note, the same case happens when someone is kicked, we should have consistent behavior there IMHO.

nbeckindy commented:
@saghul, thanks for the feedback.

The existing cancelKnocking() function calls notifyReadyToClose() on API. Is that enough, or should it send a separate event for rejection?

I added a redirect after a the kick event on desktop (native already redirects after closing the prompt). It appears the KICKED_OUT action already calls notifyKickedOut() on API so I didn't add an API event here.

In regards to consistency, I noticed that native users aren't informed when they are rejected from the lobby (they are just silently redirected) so I added an AlertDialog for that.

saghul commented:
@Calinteodor can you please review this when you get a chance?

saghul commented:
@nbeckindy We just landed some large changes were working on and there is now a small conflict. Can you please rebase your PR?

nbeckindy commented:
> @nbeckindy We just landed some large changes were working on and there is now a small conflict. Can you please rebase your PR?

No problem, should be good now

saghul commented on react/features/lobby/middleware.js:
```diff
@@ -311,6 +313,8 @@ function _conferenceFailed({ dispatch, getState }, next, action) {
             }, NOTIFICATION_TIMEOUT_TYPE.LONG)
         );
+
+        setTimeout(() => dispatch(cancelKnocking()), NOTIFICATION_TIMEOUT.LONG);
```
I think we should instead use an alert dialog with an ok button which performs the cancelKnocking upon click.

saghul commented on react/features/conference/actions.web.js:
```diff
@@ -36,5 +38,7 @@ export function notifyKickedOut(participant: Object, _: ?Function) { // eslint-d
             titleArguments: args
         }, NOTIFICATION_TIMEOUT_TYPE.STICKY));
+
+        setTimeout(() => dispatch(maybeRedirectToWelcomePage()), NOTIFICATION_TIMEOUT.LONG);
```
I think we should instead use an alert dialog with an ok button which performs the redirection upon click. We might also need to dispatch `readyToClose` I think.

saghul commented on react/features/mobile/navigation/middleware.js:
```diff
@@ -29,7 +30,12 @@ function _conferenceFailed({ dispatch }, next, action) {
+            contentKey: {
+                key: 'lobby.joinRejectedMessage'
+            },
+            onSubmit: () => dispatch(appNavigate(undefined))
```
Should we use `cancelKnocking` instead? It will clear the knocking state as well.
